src/Cake.Common.Tests/Unit/Tools/NuGet/Pack/NuGetPackerTests.cs

```diff
@@ -651,6 +652,83 @@ public void Should_Add_Verbosity_To_Arguments_If_Set(NuGetVerbosity verbosity, s
                     // Then
                     Assert.Equal(expected, result.Args);
                 }
+
+                [Fact]
+                public void Should_Add_Properties_To_Arguments_If_Set()
+                {
+                    // Given
+                    var fixture = new NuGetPackerWithProjectFileFixture();
+                    fixture.Settings.Properties = new Dictionary<string, string>
+                    {
+                        { "Configuration", "Release" }
+                    };
+
+                    // When
+                    var result = fixture.Run();
+
+                    // Then
+                    Assert.Equal("pack \"/Working/existing.csproj\" -Properties Configuration=Release", result.Args);
+                }
+
+                [Fact]
+                public void Should_Separate_Properties_With_Semicolon()
+                {
+                    // Given
+                    var fixture = new NuGetPackerWithProjectFileFixture();
+                    fixture.Settings.Properties = new Dictionary<string, string>
+                    {
+                        { "Configuration", "Release" },
+                        { "Foo", "Bar" }
+                    };
+
+                    // When
+                    var result = fixture.Run();
+
+                    // Then
+                    Assert.Equal("pack \"/Working/existing.csproj\" -Properties Configuration=Release;Foo=Bar", result.Args);
+                }
+
+                [Theory]
+                [InlineData("")]
```

Comment:
I'd have to create a separate test with a different `IDictionary<strong,string> implementation for that, or use a mock instead of a real dictionary since `Dictionary<>` already throws for null keys.